test/bats/provider.bats

```diff
@@ -30,16 +32,12 @@ setup(){
     # For AKS: "<dns-prefix>.hcp.<region>.azmk8s.io" (same as API server address, but with quotes)
     # For GKE: https://container.googleapis.com/v1/projects/<project>/locations/<az|region>/clusters/<cluster-name>
     kubectl --namespace=csi exec vault-0 -- sh -c 'vault write auth/kubernetes/config \
-        token_reviewer_jwt="$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
-        kubernetes_host="https://$KUBERNETES_PORT_443_TCP_ADDR:443" \
-        kubernetes_ca_cert=@/var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
-        issuer="https://kubernetes.default.svc.cluster.local"'
-    kubectl --namespace=csi exec vault-0 -- vault auth enable -namespace=acceptance kubernetes
-    kubectl --namespace=csi exec vault-0 -- sh -c 'vault write -namespace=acceptance auth/kubernetes/config \
-        token_reviewer_jwt="$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
-        kubernetes_host="https://$KUBERNETES_PORT_443_TCP_ADDR:443" \
-        kubernetes_ca_cert=@/var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
-        issuer="https://kubernetes.default.svc.cluster.local"'
+        kubernetes_host="https://$KUBERNETES_PORT_443_TCP_ADDR:443"'
+    if [ -n "${VAULT_LICENSE}" ]; then
+        kubectl --namespace=csi exec vault-0 -- vault auth enable -namespace=acceptance kubernetes
+        kubectl --namespace=csi exec vault-0 -- sh -c 'vault write -namespace=acceptance auth/kubernetes/config \
+            kubernetes_host="https://$KUBERNETES_PORT_443_TCP_ADDR:443"'
+    fi
```

Comment:
Does the `-namespace` option accept an empty value? Since the OSS and ENT commands are the same here, I wonder if we could set the namespace in a variable above somewhere if $VAULT_LICENSE is present, and otherwise leave it blank and run the same commands for both ENT and OSS.

Reply:
It should do, however, for the ent tests we use both mounts. Most of the SPCs refer to non-namespaced secrets and then we have one SPC that uses a Vault namespace, so this is an additional mount for that test.

I would love to do a large refactor on these tests to be smaller and easier to work with, but I would also like to make them Go-based like some of our other newer integration test suites, so I was planning to save that big refactor for doing the switch at some point.